app/models/subject.rb

```diff
@@ -0,0 +1,11 @@
+class Subject < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+  validates :code, presence: true, uniqueness: true, format:
+  {
+    with: /\A[A-Z]{2}\S{5}/,
+    message: 'has invalid format'
+  }
+  validates :year, presence: true, inclusion: 1..4
+
+  has_and_belongs_to_many :lecturers
```

Comment:
It is good practice to arrange the methods in the Model in a predictable manner (ie. all the relationships at the top, then validations, public methods, private methods)